### ISSUE TITLE
amp-img: short circuit webp fallback for iOS 

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#18599]
if `src` of an `amp-img` points to a `.webp` file, we should be able to short circuit to fallback immediately without waiting for `img` load to fail. This will remove one extra download in iOS case.

There will be cases where file extension is not present which this won't support. Also supporting `srcset` requires parsing and even in that case a mixed srcset will not work so skipping `srcset` all together for this.

There is a small chance some files with `.webp` extension are not actually `.webp` and may return some other format (maybe depending on requester) and rely on browser content-sniffing to get around the discrepancy. I think this is acceptable but we should put the new behaviour behind an experiment first just in case.

**Comment by Gregable** [comment]
There is a related old issue for this: https://github.com/ampproject/amphtml/issues/3517

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by aghassemi** [comment]
since cache server-side detects support and responds with `webp` (even for `.jpg` extension), I don't think this is something we want to do. Documenting the server-side solution (https://github.com/ampproject/amphtml/issues/21912) is the right way to go.
